extensions/_modules/boto_ec2.py

```diff
@@ -1813,7 +1813,7 @@ def detach_volume(volume_id=None, instance_id=None, device=None, force=False,
         except boto.exception.BotoServerError as e:
             log.error(e)
             return False
-    elif tags:
+    elif filters:
         try:
             volumes = conn.get_all_volumes(filters = {'tag:{0}'.format(tag_name):'{0}'.format(tag_value)})
```

Comment:
The `tag_name` and `tag_value` variables aren't getting populated anywhere in the function. Did you mean to turn that into a dictionary comprehension?